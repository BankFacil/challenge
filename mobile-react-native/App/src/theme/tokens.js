export const colors = {
	primary: '#515151',
	white: '#F7F7F7',
	border: '#BEBEBE'
};

export const space = [0, 4, 8, 16, 32, 64, 128, 256, 512];

export const radii = [0, 4, 8, 16, 32, 64, 128, 256, 512];

export const fontSizes = {
	sm: 12,
	md: 16,
	lg: 18,
};

export const fontWeights = {
	hairline: '100',
	thin: '200',
	light: '300',
	normal: '400',
	medium: '500',
	semibold: '600',
	bold: '700',
	extrabold: '800',
	black: '900',
  };
  