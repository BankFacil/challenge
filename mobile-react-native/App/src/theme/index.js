import * as tokens from './tokens';

export default {
  name: 'Github Theme',
  ...tokens,
};