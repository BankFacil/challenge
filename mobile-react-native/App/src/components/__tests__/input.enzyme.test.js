import React from 'react';
import { mount } from 'enzyme';
import { Input } from '..';
import { ThemeProvider } from 'emotion-theming';
import theme from '../../theme';

const SetupInput = props => <ThemeProvider theme={theme} {...props} />;

describe('<Input/>', () => {
  it('should render rigth', () => {
    const wrapper = mount(
      <SetupInput>
        <Input placeholder="Hi hummant, i'm a test" />
      </SetupInput>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
