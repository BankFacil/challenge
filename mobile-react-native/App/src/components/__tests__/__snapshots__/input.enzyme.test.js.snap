// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> should render rigth 1`] = `
<SetupInput>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "border": "#BEBEBE",
          "primary": "#515151",
          "white": "#F7F7F7",
        },
        "fontSizes": Object {
          "lg": 18,
          "md": 16,
          "sm": 12,
        },
        "fontWeights": Object {
          "black": "900",
          "bold": "700",
          "extrabold": "800",
          "hairline": "100",
          "light": "300",
          "medium": "500",
          "normal": "400",
          "semibold": "600",
          "thin": "200",
        },
        "name": "Github Theme",
        "radii": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
      }
    }
  >
    <Input
      onChangeText={[Function]}
      placeholder="Hi hummant, i'm a test"
    >
      <emotion(TextInput)
        borderColor="border"
        borderRadius={1}
        borderWidth={1}
        color="primary"
        fontSize="sm"
        fontWeight="bold"
        height={30}
        onChangeText={[Function]}
        placeholder="Hi hummant, i'm a test"
        placeholderTextColor="#515151"
        py="0"
        textAlign="center"
      >
        <TextInput
          allowFontScaling={true}
          borderColor="border"
          borderRadius={1}
          borderWidth={1}
          color="primary"
          fontSize="sm"
          fontWeight="bold"
          height={30}
          onChangeText={[Function]}
          placeholder="Hi hummant, i'm a test"
          placeholderTextColor="#515151"
          py="0"
          rejectResponderTermination={true}
          style={
            Object {
              "borderColor": "#BEBEBE",
              "borderRadius": 4,
              "borderWidth": 1,
              "color": "#515151",
              "fontSize": 12,
              "fontWeight": "700",
              "paddingBottom": 0,
              "paddingTop": 0,
              "textAlign": "center",
            }
          }
          textAlign="center"
          underlineColorAndroid="transparent"
        >
          <TextInput
            allowFontScaling={true}
            borderColor="border"
            borderRadius={1}
            borderWidth={1}
            color="primary"
            fontSize="sm"
            fontWeight="bold"
            height={30}
            onChangeText={[Function]}
            placeholder="Hi hummant, i'm a test"
            placeholderTextColor="#515151"
            py="0"
            rejectResponderTermination={true}
            style={
              Object {
                "borderColor": "#BEBEBE",
                "borderRadius": 4,
                "borderWidth": 1,
                "color": "#515151",
                "fontSize": 12,
                "fontWeight": "700",
                "paddingBottom": 0,
                "paddingTop": 0,
                "textAlign": "center",
              }
            }
            textAlign="center"
            underlineColorAndroid="transparent"
          />
        </TextInput>
      </emotion(TextInput)>
    </Input>
  </ThemeProvider>
</SetupInput>
`;
